feat(components): Add clearable to DataList Search

rodrigoeidelvein opened the pull request:
<!--
  Atlantis uses Conventional Commits to track versions.
  Pull request titles should follow the following format.

  For help creating your pull request, you can [use this tool](https://atlantis.getjobber.com/?path=/story/guides-pull-request-title-generator--page)

  <TYPE>(<optional SCOPE>): <conditionally BREAKING CHANGE:> <description>

  eg.
    fix(SCOPE): stop graphite breaking when too much pressure applied — Patch Release
    feat(SCOPE): add 'graphiteWidth' option — (Minor) Feature Release
    feat(SCOPE): BREAKING CHANGE: remove graphiteWidth option — (Major) Breaking Release

  TYPE should consist of:
    - fix: a commit of the type fix patches a bug in your codebase
    - feat: a commit of the type feat introduces a new feature to the codebase
    - docs: documentation only changes
    - build: improvements to the build system
    - refactor: a change that neither fixes a bug nor introduces a feature
    - chore: other changes that don't modify src or test files

  SCOPE should be one of:
    - components
    - components-native
    - design
    - eslint
    - generators
    - hooks
    - stylelint


  If your pull request introduces a breaking change please append `BREAKING CHANGE:` following type / scope.

  Further Reading:
    - https://www.conventionalcommits.org
    - https://github.com/commitizen/conventional-commit-types/blob/master/index.json
-->

## Motivations

Making it easier for SPs to clear the search on DataList. 

I had to add a `max-width` to the input search container because it increased the width when the clear button was being shown. I set it to `base * 13` (208px) because it's closer to the current width that was being rendered (214px). I'm open to suggestions if you know a better solution.

## Changes

<!-- https://keepachangelog.com/en/1.0.0/ -->

### Added

- <!-- new features -->

### Changed

- <!-- changes in existing functionality -->

### Deprecated

- <!-- soon-to-be removed features -->

### Removed

- <!-- now removed features -->

### Fixed

- <!-- for any bug fixes -->

### Security

- <!-- in case of vulnerabilities -->

## Testing

<!-- How to test your changes. -->

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

![Random photo of Atlantis](https://loremflickr.com/672/400/atlantis)

chris-at-jobber commented on packages/components/src/DataList/components/DataListSearch/DataListSearch.css:
```diff
@@ -32,6 +32,7 @@
     position: static;
     visibility: visible;
     width: auto;
+    max-width: calc(var(--base-unit) * 13);
```
Isn't the row that holds the search using `<Grid>`? I would expect this search input to take up the prescribed column value rather than be responsible for its' own width

rodrigoeidelvein replied:
We're not using `Grid` here. But we are using `display: grid;` [here](https://github.com/GetJobber/atlantis/blob/080c07cb01bf6842de106c51f167120ec2a168a9/packages/components/src/DataList/DataList.css#L32), where I could set `grid-template-columns: auto fit-content(208px);` and it would work too.

chris-at-jobber replied:
Oh OK - I'm not super opinionated about how we do it, but it does seem like we might want to put the layout responsibilities outside of this component.

chris-at-jobber commented:
Is it possible to get the clear button closer to the right of the input when it's small?

![image](https://github.com/GetJobber/atlantis/assets/39704901/90531d51-8977-4b41-a0ff-bd27b0d9fd96)

Also, I just went back and looked at the original mockups for this list, and I think this should be using `base` size Input:

![image](https://github.com/GetJobber/atlantis/assets/39704901/f9e05843-a36d-44e7-86e8-e429b6ffa555)

rodrigoeidelvein commented:
@chris-at-jobber I changed the input to `base`, now it shows the mini label when there's a value. Is that okay? I also moved the clear button a little bit to the left of the input and handled the max-height in the grid container

chris-at-jobber commented:
Looks good! Just need a smidgen more padding when it collapses for small screens:
![image](https://github.com/GetJobber/atlantis/assets/39704901/e1f3ec95-f635-4d8f-8311-87ecc00b0fef)

Currently it's `small`, if we go up to something like `small` + `smaller` we should be good:
![image](https://github.com/GetJobber/atlantis/assets/39704901/4240170c-1179-4d60-aa11-d0c0362b1986)
![image](https://github.com/GetJobber/atlantis/assets/39704901/2dbd68dd-d36e-42de-a83a-eeb4f1eeab32)

chris-at-jobber reviewed (changes requested):
Just this change [here](https://github.com/GetJobber/atlantis/pull/1697#issuecomment-1865260006) and we're good
